File: empty-shell-with-agentcore-sdk/invoker.py
import json
from pathlib import Path
import boto3

client = boto3.client("bedrock-agentcore")

runtime_arn = Path("tmp/agent_runtime_arn.txt").read_text().strip()

print(f"> runtime_arn={runtime_arn}")

payload = {"input": {"prompt": "hello"}}
payload = json.dumps(payload)
print(f"> payload={payload}")

print(f"> invoking...")
response = client.invoke_agent_runtime(
    agentRuntimeArn=runtime_arn,
    payload=payload,
)

print(f"> response: {response}")
body = response["response"].read().decode()
print(f"> body: {body}")

File: empty-shell-with-expressjs/invoker.js
import { readFileSync } from 'fs';
import { BedrockAgentCoreClient, InvokeAgentRuntimeCommand } from '@aws-sdk/client-bedrock-agentcore';

const runtimeArn = readFileSync('./tmp/agent_runtime_arn.txt', 'utf-8').trim();
console.log(`> RUNTIME_ARN=${runtimeArn}`);

const client = new BedrockAgentCoreClient();
const payload = JSON.stringify({ hello: 'world' });

console.log('> Invoking...');
const command = new InvokeAgentRuntimeCommand({
  agentRuntimeArn: runtimeArn,
  payload: Buffer.from(payload),
  contentType: 'application/json',
});

const response = await client.send(command);
const responseString = await response.response.transformToString()

console.log(responseString);

